lib/OpenQA/Git.pm

```diff
@@ -3,29 +3,25 @@
 
 package OpenQA::Git;
 
-use Mojo::Base -base;
+use Mojo::Base -base, -signatures;
 use Cwd 'abs_path';
 use OpenQA::Utils qw(run_cmd_with_log_return_error);
 
 has 'app';
 has 'dir';
 has 'user';
 
-sub enabled {
-    my ($self) = @_;
+sub enabled ($self, $args = undef) {
     die 'no app assigned' unless my $app = $self->app;
     return ($app->config->{global}->{scm} || '') eq 'git';
 }
 
-sub config {
-    my ($self) = @_;
+sub config ($self, $args = undef) {
     die 'no app assigned' unless my $app = $self->app;
     return $app->config->{'scm git'};
 }
 
-sub _prepare_git_command {
-    my ($self, $args) = @_;
-
+sub _prepare_git_command ($self, $args = undef) {
     my $dir = $args->{dir} // $self->dir;
```

Comment:
```suggestion
    my $dir = $args ? $args->{dir} : undef // $self->dir;
```

Reply:
it's actually not problematic to do `$undefined->{key}`.
The variable will then be autovivified to a hash:
```
my $default = 23;
my $args = undef;
say ref $args;
my $dir = $args->{dir} // $default;
say $dir;      
say ref $args;
__END__

23
HASH
```

Reply:
Ah, right. So that's why we didn't notice any problems with it. I suppose the code is ok then.